Sources/Datadog/Logger.swift

```diff
@@ -80,9 +80,12 @@ public class Logger {
     private var loggerAttributes: [String: Encodable] = [:]
     /// Taggs associated with every log.
     private var loggerTags: Set<String> = []
+    /// Queue ensuring thread-safety of the `Logger`. It synchronizes tags and attributes mutation.
+    private let queue: DispatchQueue
 
-    init(logOutput: LogOutput) {
+    init(logOutput: LogOutput, queue: DispatchQueue) {
```

Comment:
as `queue` is an implementation detail in `Logger`, i think it should not be exposed in any way.

because
• interference from outside may break its function (ie: sync'ing read-write within `Logger`)
• ideally, we need to test all possible `DispatchQueue` configurations

so, i propose 
• passing `identifier: String` to `init` and 
• adding `var identifier { queue.identifier }` public computed property

```swift
var identifier: String { queue.label }
...
init(identifier: String, ...) {
  queue = DispatchQueue(label: identifier, qos: .userInteractive)
  ...
}
```

Reply:
Indeed, `DispatchQueue` dependency is "too wide" and we can narrow it with an "identifier".

However, I don't see the reason to expose public `identifier: String` from the `Logger` - why this might be useful for the user? As a rule of thumb 👍 - the less things we expose as `public` the better, as it means less hassle with backward compatibility.

Changes pushed 🚀, let me know WDYT.

Reply:
super 💯 

me neither, i don't see how it might be useful; as another rule of thumb, i think the user should get what they give.
but you are right, backward compatibility overshadows my argument 👍 

then `label` (or `queueLabel` even better?) sounds better than `identifier` as the user cannot identify `Logger` with it?

Reply:
> then, wouldn't label (or even queueLabel?) sound better than identifier as the user cannot identify Logger with it?

I like the "identifier" as it somehow highlights the behavioral usage of this variable (it's to identify the logger). We might also want to use it for other things in the future. Anyway, IMO it's good enough 💪.